### ISSUE TITLE
Loading dialog before page is truly loaded fails

### ISSUE THREAD

**Issue by bfattori** [NativeScript/NativeScript#1516]
I have an app which requires a login. To accomplish this, I have a login page, with a model, that loads and then executes a method in the login-page-model.js file called `login()`:

```
var lpModule = require("./models/login-page-model");

function pageLoaded(args) {
    var page = args.object;
    page.bindingContext = lpModule.loginModel;

    lpModule.loginModel.login();
}
exports.pageLoaded = pageLoaded;
```
The method being called is this:
```
    LoginModel.prototype.login = function() {
        // This is not secure (does it matter?)
        var uName = this.get("lastUser"),
            _this = this;

            dialogs.login({
                title: "Process Control",
                message: "Enter your login information",
                okButtonText: "Log In",
                userName: uName
            }).then(function(res) {
                _this.doLogin(res)
            });
    };
```
The page displays, but the dialog _never displays_ on Android. On iOS, this works fine. To make it work on Android, I had to put a timeout around the call to the `dialogs.login`, e.g:
```
        setTimeout(function() {
            dialogs.login({
                title: "Process Control",
                message: "Enter your login information",
                okButtonText: "Log In",
                userName: uName
            }).then(function(res) {
                _this.doLogin(res)
            });
        }, 500);
```
Is there something that I'm assuming wrong? Is the page not truly loaded when the callback is called? Is there a way to know that the application is _ready_?

**Comment by NathanaelA** [comment]
Have you tried to put this in "onNavigatedTo" that event occurs after everything and do to some changes in v1.6.0 the pageLoaded is NOT recommended for do dialogs in iOS or Android.

**Comment by bfattori** [comment]
Is this in the docs? I didn't see that.

Brett

**Comment by NathanaelA** [comment]
I'm sure all the different navigation events are somewhere in the docs.    But I also have a blog post on all the different page/modal events and when they are fired for easy reference. (http://fluentReports.com/blog).  

And the reason I know about that pageLoaded will change in 1.6 is because I have a different bug in on dialogs that they did a specific fix in v1.5 (which is why iOS actually works right now), but they are going to revert that patch and make Android actually match iOS; so that dialogs will only work on the onNavigatedTo event now because of page transitions.
